### PR TITLE
feat: New tabs

### PR DESCRIPTION
**Pull request by zombieJ**
还需要改造支持移动设备滚动，交互可以先看起来。

ref https://github.com/ant-design/ant-design/issues/24284

Preview：https://tabs-git-new-tabs.react-component.now.sh/iframe.html?id=rc-tabs--mix

- [x] Hover to open
- [x] `...` follow the last tab

### CODE REVIEW

**Comment by zombieJ** [comment]
奇怪，now 怎么不触发了

**Comment by zombieJ** [comment]
More 跟随最后一个可见 Tab 位置很不稳定，不如固定位置好。

File: src/Tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+  // Reset active key if not exist anymore
+  useEffect(() => {
+    let newActiveIndex = tabs.findIndex(tab => tab.key === mergedActiveKey);
+    if (newActiveIndex === -1) {
+      newActiveIndex = Math.max(0, Math.min(activeIndex, tabs.length - 1));
+      setMergedActiveKey(tabs[newActiveIndex]?.key);
+    }
+    setActiveIndex(newActiveIndex);
+  }, [tabs.map(tab => tab.key).join('_'), mergedActiveKey, activeIndex]);
```
Comment (Tav0) [inline src/Tabs.tsx:140]: This is a potential bug, it is creating a new array every time thus triggering this useEffect more than needed

Comment (zombieJ) [reply src/Tabs.tsx:140]: array is `join` to string, not count array as condition.

**Pull request merged**